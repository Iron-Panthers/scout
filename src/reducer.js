export const reducer = (state, action) => {
  switch (action.type) {
    case "next_mode":
      const modes = [ "Configure", "Scout", "Review", "Scan" ]
      return {
        ...state,
        mode: modes[modes.indexOf(state.mode) + 1]
      }
    // base reducer, no special behavior
    case "set":
      console.log(action.val)
      return {
        ...state,
        [action.prop]: action.val
      }
    default:
      return state
  }
}

const shooting = {
  innerOuterSucc: 0,
  innerOuterFail: 0,
  lowerSucc: 0,
  outerSucc: 0
}

export const initialState = {
  mode: "Configure",
  team: undefined,
  matchType: "Test",
  matchNum: undefined,
  timeLeft: 150, //seconds
  phase: "auto", //auto, teleop, endgame
  auto: {
    pathType: undefined,
    ...shooting
  },
  tele: {
    ...shooting
  },
  end: {
    climb: false,
    park: false,
    level: false, //level vs not level, not level is false
    levelTime: undefined //value of timeLeft when level is set
  },
  underTrench: false,
  defense: false,
  problems: false,
  comments: "",
}