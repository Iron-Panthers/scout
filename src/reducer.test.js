import { initialState } from "./reducer.js"

it("has an initial state that matches the snapshot", () => {
  expect(JSON.stringify(initialState)).toMatchSnapshot()
})
