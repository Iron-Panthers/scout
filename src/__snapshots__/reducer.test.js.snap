// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has an initial state that matches the snapshot 1`] = `"{\\"version\\":1,\\"mode\\":\\"Configure\\",\\"matchType\\":\\"Qualification\\",\\"phase\\":\\"auto\\",\\"auto\\":{\\"pickup\\":0,\\"taxi\\":false,\\"upperSucc\\":0,\\"upperFail\\":0,\\"lowerSucc\\":0,\\"lowerFail\\":0},\\"teleop\\":{\\"upperSucc\\":0,\\"upperFail\\":0,\\"lowerSucc\\":0,\\"lowerFail\\":0},\\"endgame\\":{\\"climb\\":false,\\"fail\\":false,\\"level\\":\\"None\\"},\\"wrongCargo\\":false,\\"defense\\":false,\\"scoutProblems\\":false,\\"robotProblems\\":false,\\"comments\\":\\"\\",\\"undoStack\\":{\\"auto\\":[],\\"teleop\\":[],\\"endgame\\":[]}}"`;
